examples/functions-domain/functions.dsl

```diff
@@ -0,0 +1,28 @@
+type Set
+type Point
+type Map
+
+-- NOTE: we are not porting all the predicates on Intervals and Functions into this domain, as we are only drawing continuous functions and bounded, connected intervals.
+type Interval
+type Function
+type Reals
+type Real
+Reals <: Set
+Interval <: Set
+Reals <: Interval
+Function <: Map
+
+constructor CreateInterval: Real left * Real right -> Interval
+constructor CreateFunction: Set domain * Set codomain -> Function
+
+predicate Not: Prop p1 -- higher-order predicate 
+
+predicate From : Map f * Set domain * Set codomain
+predicate In : Point p * Set s
+predicate OneToOne : Map m
+predicate Onto : Map m
+predicate Bijection : Map m
+predicate PointIn : Set s * Point p
```

Comment:
I noticed that the Alloy plugin generates `PointIn` statements. Can we generate `In` statements instead and remove `PointIn` from the Domain program? (`In` is more concise and has a more natural order of args)

Reply:
Done.